src/pybind/mgr/cephadm/ssh.py

```diff
@@ -158,7 +158,13 @@ async def _execute_command(self,
                                ) -> Tuple[str, str, int]:
 
         conn = await self._remote_connection(host, addr)
-        sudo_prefix = "sudo " if self.mgr.ssh_user != 'root' else ""
+        if self.mgr.ssh_user != 'root':
+            if self.mgr.sudo_password:
+                sudo_prefix = f"sshpass -p {self.mgr.sudo_password} sudo "
```

Comment:
I don't know if you noticed my earlier comment but I don't think this is the approach you should take. Putting the password directly into the command line could end up decreasing overall security, by making the passwords visible in /proc, or ending up in log files, precisely for the customers that are trying to put stricter password policies in place. Did you see my suggestion with regards to SUDO_ASKPASS?

Reply:
ops, this got missed.
a bit more finding, as sudo manpage says: 
                 If the -A (askpass) option
                 is specified, a (possibly graphical) helper program is
                 executed to read the user's password and output the
                 password to the standard output.
the helper program need to put the password in std output. It looks what is been avoided here?

Moreover, sshpass's -f can get a file input with password in it. It could do the same thing without password exposure in command line. How about it?

Reply:
I think investigating `sshpass -f` is worthwhile if it avoids the issue here. Do note that I also recommend using a named pipe and not a regular file so that the password data is never witten to disk, and that the named pipe be as secure as possible.

One remaining downside to `sshpass` is that it is an extra dependency to install, correct? Whereas sudo's askpass feature is built in to sudo. Note that the helper program doesn't have to be graphical it's just an example of a common use case. You also wrote "It looks what is been avoided here?" but I'm sorry to say I don't know what you mean by this.  If you're trying to say that you're trying to avoid putting a password on stdin, please note that stdout->stdin from the askpass program to sudo is only the sudo side of the interface. The askpass program could (for example) query the mgr directly or read from a named pipe and output on stdout. Here' an example in bash:
```bash
# create a named pipe
mkfifo /run/user/1000/foobar

# write some data
echo testabc123 > /run/user/1000/foobar &

# read from pipe (this is what your askpass script could do)
cat /run/user/1000/foobar
```

This is just an example, but I hope it illustrates what I'm suggesting.

Reply:
I also suggest reading the sudo docs carefully in case I missed something and there's a direct way to read the password from the pipe. However, if not my suggestion above could still serve the suggested purpose.

Reply:
like the idea of having least dependencies(sshpass). So, one askpass helper function could be added, something like "ceph cephadm askpass-helper" to put the password into stdout. However ceph common utility need to be present on each hosts of cluster. It looks like clean. What do you think of it?

Reply:
> like the idea of having least dependencies(sshpass). So, one askpass helper function could be added, something like "ceph cephadm askpass-helper" to put the password into stdout.

Yeah, that's a very appealing idea.

> However ceph common utility need to be present on each hosts of cluster. It looks like clean. 

cephadm already makes a copy of cephadm on every host it needs to run cephadm on. So you are good there

> What do you think of it? @phlogistonjohn

It's appealing however there are a few things that we need to watch out for. First and foremost is that you need to run the command as non-root. This may need additional changes to cephadm but that's not a big change and I have already made similar changes myself. But, depending on what you intend for that command to do, you'll need to stick to simple actions that are built into python because any of the fancy stuff that cephadm does, it does inside containers and those containers are (generally) only started after becoming root. If you're hoping to make ceph api calls from cephadm directly that probably won't work though.

One minor additional thought, in my pr #51738 I'm adding a new "namespace" to the cephadm cli called "_orch". Future commands will look like `cephadm _orch deploy` or `cephadm _orch frobnicate`. Things following "_orch" are meant to not be run by humans but initiated by the mgr.   I'm not asking you to make any changes with regards to this now, but if my PR gets merged before this on it might be good to put the new command in the "_orch" namespace because it's not meant to be run directly by people.